### PR TITLE
YD-174 More HATEOAS

### PR DESCRIPTION
**Pull request by Bert-R**
- Added more collection links (e.g. messages)
- Made the tests use the links instead of URL composition.

### CODE REVIEW

File: appservice/src/main/java/nu/yona/server/analysis/rest/AppActivityController.java [hunk 1]
```diff
@@ -52,4 +58,28 @@ public void addAppActivity(@RequestHeader(value = PASSWORD_HEADER) Optional<Stri
...
 		});
 	}
+
+	@RequestMapping(value = "/", method = RequestMethod.GET)
+	@ResponseBody
+	@ResponseStatus(HttpStatus.OK)
+	public HttpEntity<AppActivityDTO[]> getAppActivities(@RequestHeader(value = PASSWORD_HEADER) Optional<String> password,
+			@PathVariable UUID userID)
+	{
+		// This method is just a shorthand to facilitate link building
```
Comment (aristotelos) [inline appservice/src/main/java/nu/yona/server/analysis/rest/AppActivityController.java]: I don't think this is necessary or good to do. The intention is to share a link to which the user can POST app activities. Therefore the link can use the `addAppActivity` method and the `getAppActivities` method is not necessary.

File: appservice/src/main/java/nu/yona/server/analysis/rest/AppActivityController.java [hunk 2]
```diff
@@ -52,4 +58,28 @@ public void addAppActivity(@RequestHeader(value = PASSWORD_HEADER) Optional<Stri
...
+		return null;
+	}
+
+	public static Link getAppActivityLink(UUID userID)
+	{
+		try
+		{
+			ControllerLinkBuilder linkBuilder = linkTo(
+					methodOn(AppActivityController.class).getAppActivities(Optional.empty(), userID));
```
Comment (aristotelos) [inline appservice/src/main/java/nu/yona/server/analysis/rest/AppActivityController.java]: Please use `addAppActivity` method with third parameter `null`.

File: appservice/src/main/java/nu/yona/server/analysis/rest/AppActivityController.java (hunk 1 above)
Comment (Bert-R) [inline appservice/src/main/java/nu/yona/server/analysis/rest/AppActivityController.java]: I tried that, but that doesn't work.

File: appservice/src/main/java/nu/yona/server/analysis/rest/AppActivityController.java (hunk 2 above)
Comment (Bert-R) [inline appservice/src/main/java/nu/yona/server/analysis/rest/AppActivityController.java]: See previous comment.

File: core/src/testUtils/groovy/nu/yona/server/YonaServer.groovy [hunk 3]
```diff
@@ -39,98 +30,6 @@ class YonaServer
 		formatter.format(new Date())
 	}
 
-	def deleteGoal(goalURL)
```
Comment (aristotelos) [inline core/src/testUtils/groovy/nu/yona/server/YonaServer.groovy]: Nice cleanup :)

**Pull request merged**